### ISSUE TITLE
Run GLUE for fine-tuning Few-Shot Relation Classification

### ISSUE THREAD

**Issue by fillipefbr** [osainz59/Ask2Transformers#4]
Hi, 
Thank you for this amazing repository, it is exactly what I was looking for! And congratulations for this work :)

I have two doubts about the fine-tuning process for Few-Shot RC:

**1.** I successfully used the provided script to convert TACRED data to MNLI, however, the [run_glue.py](https://github.com/osainz59/Ask2Transformers/blob/master/a2t/relation_classification/run_glue.py) script takes too long to train (I'm using Colab and it's taking like 23 hours per epoch, with a huge number of examples and optimization steps even with tiny splits of the dataset ). Am I missing something? These are the parameters I used:

`python Ask2Transformers/a2t/relation_classification/run_glue.py --task_name mnli --train_file train.mnli.json --validation_file dev.mnli.json --test_file test.mnli.json --model_name_or_path roberta-large-mnli --output_dir output --cache_dir cache --do_train --do_eval --do_predict --seed 6 --per_device_train_batch_size 16 --overwrite_output_dir`

**2.**  After running the [run_glue.py](https://github.com/osainz59/Ask2Transformers/blob/master/a2t/relation_classification/run_glue.py) script, is the fine-tuned model supposed to be found in the output _(--output_dir)_ directory? (I'm using hugging face transformers). Sorry if this question seems dumb, I'm just not sure how to proceed and use the fine-tuned model after training. 

I would appreciate if you could give me some guidance about this process.
Thank you!

**Comment by osainz59** [comment]
Hi!
Thanks for your comments I really appreciate them!

Regarding the question about the time required for training the model, yes, fine-tuning a model with this method takes time. In fact, it is proportional to the number of entailment/neutral/contradiction examples generated per TACRED example. I can give you some approximations based on my hardware:

GPU: 2080ti (11Gb)

| Hyperparameter | Value |
| :-----------------|:------|
| posn | 2 |
| negn | 5 |
| epochs | 5 |
| per_device_train_batch_size | 16 |
| gradient_accumulation | 2 |
| fp16 | True | 

With that configuration I am able to train `roberta-large` at **2.86 hours/epoch** (whole dataset). Nevertheless, the results reported at the paper were obtained using `posn` and `negn` equal to 1, so, the training should be faster (check the [paper ](https://aclanthology.org/2021.emnlp-main.92.pdf) Appendix B for more details).

Regarding the question about the output directory, also yes. Once you train your model you can pass the path to the `pretrained_model` argument to load and test your fine-tuned model. I was planning to upload already trained models to the HuggingFace Hub, but I am quite busy right now.

Let me know if you have any other question :)

**Comment by fillipefbr** [comment]
Thank you very much!

I actually found out that the problem was related to the `--task_name mnli` parameter, which was ignoring my local `train/dev/test` files. Removing this parameter made everything work as expected 🎉
